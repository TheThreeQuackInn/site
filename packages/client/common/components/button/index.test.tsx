import React from 'react';
import {create} from 'react-test-renderer';
import Button from './';

describe('Button', () => {
    it('renders the primary variant correctly', () => {
        const onClick = jest.fn();
        const component = create(<Button label="Testing" onClick={onClick} variant="primary" />);
        expect(component).toMatchSnapshot();
    });

    it('renders the outline variant correctly', () => {
        const onClick = jest.fn();
        const component = create(<Button label="Testing" onClick={onClick} variant="outline" />);
        expect(component).toMatchSnapshot();
    });
});
