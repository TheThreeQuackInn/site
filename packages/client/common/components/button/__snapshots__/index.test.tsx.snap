// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders the outline variant correctly 1`] = `
<button
  className="rounded focus:outline-black dark:focus:outline-white bg-transparent hover:bg-transparent dark:text-white"
  onClick={[MockFunction]}
  type="button"
>
  Testing
</button>
`;

exports[`Button renders the primary variant correctly 1`] = `
<button
  className="rounded focus:outline-black dark:focus:outline-white text-white button-gradient"
  onClick={[MockFunction]}
  type="button"
>
  Testing
</button>
`;
